1525/listing export take 2

ColinBuyck opened the pull request:
# Pull Request Template

## Issue Overview

This PR addresses #issue

- [ ] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Please include a summary of the change and which issue(s) is addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## How Can This Be Tested/Reviewed?

Provide instructions so we can review.

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

ColinBuyck commented on backend/core/src/listings/listings-csv-exporter.service.ts:
```diff
@@ -0,0 +1,163 @@
+        Digital_Application: listing.digitalApplication,
+        Digital_Application_URL: listing.applicationMethods[1]?.externalReference,
+        Paper_Application: listing.paperApplication,
+        Paper_Application_Filename:
```
Changed from Paper_Application_URL since Paper Application doesn't allow for url option but still confirming change with product

ColinBuyck replied:
Clarified by product!

ColinBuyck commented on backend/core/src/listings/listings-csv-exporter.service.ts:
```diff
@@ -0,0 +1,163 @@
+        Lottery_End: dayjs(listing.events?.endTime).format("MM-DD-YYYY h:mm:ssA"),
+        Lottery_Notes: listing.events[0]?.note,
+        Application_Due_Date: listing.applicationDueDate,
+        Waitlist: listing.isWaitlistOpen,
```
Confirming with product if this is the correct field or if its intended to be "displayWaitlist"

emilyjablonski reviewed (changes requested):
I've found a few data issues in testing:
Listing CSV
* Building selection criteria is empty even if I have uploaded something
* Lottery date, lottery start, and lottery end are empty even if they have data
* Street1 for leasing agent pickup address is empty even if it has data
* If I upload a high quality PDF as a paper application, the link in the spreadsheet is to a low res jpg image of just the first page
* If there is data in "is there an application due date" nothing shows up in the column. To me it seems like this question should toggle a date field but it isn't, I'm checking in [slack](https://exygy.slack.com/archives/C02G1S19QA2/p1678728712555439), so either this cell needs a "yes/no" or a date pending the resolution.

Units CSV
* In a unit group I am able to add multiple AMI levels but only one is appearing in the CSV
* If the rent type is percentage of income, the rent cell is empty

Visually it also looks like the tab order is funky. It might be hitting the button twice and the visual focus states are either not appearing or misaligned. When it comes to the listing button that might be existing.

I think it would be nice to have the dates formatted more consistently / readable. I like how the postmark field is displayed as `10-10-2023 10:00:00PM` for example but other dates are formatted like this: `Mon Mar 13 2023 11:14:04 GTM-0600 (Mountain Daylight Time)` in a way where it's showing an "incorrect" time.

I also don't know if this is within scope but I'm also generally wishing the data was formatted less like the way the data is stored, I worry it might be confusing or just less crisp, ie:
* The listing table in partners has a `Listing Status` column and the displayed values there are `Public` and `Draft` but the CSV shows `active` and `pending`
* Keys coming through, like in a11y features like `wheelchairRamp`, review order as `firstComeFirstServe`, marketing status as `comingSoon` etc

emilyjablonski commented on backend/core/src/listings/listings-csv-exporter.service.ts:
```diff
@@ -0,0 +1,179 @@
+    })
+
+    const listingObj = listings.map((listing) => {
+      console.log(listing.name)
```
🧹

emilyjablonski commented:
![Screenshot 2023-03-13 at 12 23 36 PM](https://user-images.githubusercontent.com/65367387/224797810-022432ac-b174-45fe-9eaf-b4df1bf930e9.png)
One more visual note - looks like the success toast is not aligned with the content. This update to that alert should do the trick:

`<div className="flex z-50 flex-col items-center justify-end">
            <SiteAlert
              dismissable
              timeout={5000}
              alertMessage={{ message: t("listings.exportSuccess"), type: "success" }}
              sticky
              className={"mr-0"}
            />
          </div>`

ColinBuyck commented:
Thank you for the detailed review @emilyjablonski 🗻  I believe I have addressed all of the comments above with the exception of the toast piece since we decided to maintain alignment with the User Export. Let me know what you think!

emilyjablonski commented on backend/core/src/listings/helpers.ts:
```diff
@@ -0,0 +1,88 @@
+import dayjs from "dayjs"
```
🥳

emilyjablonski reviewed (approved):
🐭 🕳️

ludtkemorgan commented on sites/partners/src/pages/index.tsx:
```diff
@@ -174,11 +203,24 @@ export default function ListingsList() {
-                    <Button className="mx-1" onClick={() => false}>
-                      {t("listings.addListing")}
+                  <div className="flex-row">
+                    <LocalizedLink href={`/listings/add`}>
```
I don't know if I'm looking at the right figma doc. But looks like between this, the user export, and the design we have slightly different looks of the buttons. The most prominent is the size of the buttons and first button not being white.
This:
![image](https://user-images.githubusercontent.com/42942267/225725846-33a1ce30-8393-40b9-8d88-dd5e1622fd63.png)

User export:
![image](https://user-images.githubusercontent.com/42942267/225725960-7fd6dea0-df45-446e-9085-f15aaf5d8452.png)

Design:
![image](https://user-images.githubusercontent.com/42942267/225726052-4956965f-e49c-427f-ab0f-462f5b139d2f.png)
I'm curious if we have a final design pattern that we should be doing. Either updating it here or on the user export

ColinBuyck replied:
Good catch! The button color was just a miss on my part so that should be aligned with the Figma now. In terms of the button sizing, the button padding in both the Figmas (units and listings) aligns with the base button size as opposed to the small one used in the User PR. If it works for you, I can align the user button size with what is currently on listings. Or I could touch base with design if you'd prefer that @ludtkemorgan ?

ludtkemorgan replied:
I see. I think my confusion was around the button being bigger than the search bar since the designs have both of them the same height. I think it makes sense to keep it bigger and update the user button. Then we can get input from design on the search bar. Sorry for the confusion, I should have noticed it my user export

ludtkemorgan commented on backend/core/src/listings/helpers.ts:
```diff
@@ -0,0 +1,88 @@
+  return value ? `$${value}` : ""
+}
+
+export const convertToTitleCase = (value: string): string => {
```
💯

ludtkemorgan reviewed (changes requested):
Functionally it all seems good. Just one design question.

Also, I think before we release this we might want to cleanup our prod users. Using Prod data all of the listings start with this for the last column
![image](https://user-images.githubusercontent.com/42942267/225733146-b3a8a999-5a5e-4e4d-87ff-4f8375a72d4f.png)
